devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusAddExtension.java

```diff
@@ -40,9 +57,16 @@ public void addExtension() {
                 .collect(toSet());
 
         try {
-            new AddExtensions(getQuarkusProject())
-                    .extensions(extensionsSet)
-                    .execute();
+            AddExtensions addExtensions = new AddExtensions(getQuarkusProject())
+                    .extensions(extensionsSet);
+            if (registry != null) {
+                Registry registry = new URLRegistryBuilder()
+                        .addURL(new URL(getRegistry()))
+                        .build();
+                DefaultExtensionRegistry extensionRegistry = new DefaultExtensionRegistry(registry);
+                addExtensions.extensionRegistry(extensionRegistry);
```

Comment:
Could we move the `Registry` and `DefaultExtensionRegistry` creation  to the `devtools-common`?

Reply:
Good point, but I'd rather have a static `fromURLs` method in `DefaultExtensionRegistry`.